pkg/ansible/runner.go

```diff
@@ -0,0 +1,201 @@
+	cfgmap := &kapi.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: inventoryConfigMap,
+		},
+		Data: map[string]string{
+			"hosts": inventory,
+			"vars":  vars,
+		},
+	}
+	cfgMap, err := r.KubeClient.CoreV1().ConfigMaps(operatorNamespace).Create(cfgmap)
+	if err != nil && kapierrors.IsAlreadyExists(err) {
+		// Update existing configmap if it already exists:
+		log.WithField("configmap", cfgMap.Name).Infoln("already exists, attempting update")
+		cfgMap, err = r.KubeClient.CoreV1().ConfigMaps(operatorNamespace).Update(cfgmap)
+		if err != nil {
+			return cfgMap, err
+		}
+	} else if err == nil {
+		log.WithField("configmap", cfgMap.Name).Infoln("created successfully")
+	}
+
+	return cfgMap, err
+}
+
+func (r *ansibleRunner) RunPlaybook(playbook string, inventory string, vars string) error {
+
+	// TODO: defer deletion for cleanup
+	_, err := r.createInventoryConfigMap(inventory, vars)
+	if err != nil {
+		return err
+	}
+
+	jobName := "openshift-ansible-test-job"
+	env := []kapi.EnvVar{
+		{
+			Name:  "INVENTORY_FILE",
+			Value: "/ansible/inventory/hosts",
+		},
+		{
+			Name:  "PLAYBOOK_FILE",
+			Value: playbook,
+		},
+		{
+			Name:  "ANSIBLE_HOST_KEY_CHECKING",
+			Value: "False",
+		},
+		{
+			Name:  "OPTS",
+			Value: "-vvv --private-key=/ansible/ssh/privatekey.pem -e @/ansible/inventory/vars",
+		},
+		{
+			Name: "AWS_ACCESS_KEY_ID",
+			ValueFrom: &kapi.EnvVarSource{
+				SecretKeyRef: &kapi.SecretKeySelector{
+					LocalObjectReference: kapi.LocalObjectReference{Name: awsCredentialsSecret},
+					Key:                  "aws_access_key_id",
+				},
+			},
+		},
+		{
+			Name: "AWS_SECRET_ACCESS_KEY",
+			ValueFrom: &kapi.EnvVarSource{
+				SecretKeyRef: &kapi.SecretKeySelector{
+					LocalObjectReference: kapi.LocalObjectReference{Name: awsCredentialsSecret},
+					Key:                  "aws_secret_access_key",
+				},
+			},
+		},
+	}
+	runAsUser := int64(0)
+	sshKeyFileMode := int32(0600)
+	playbookPath := path.Join(openshiftAnsibleContainerDir, playbook)
+	podSpec := kapi.PodSpec{
+		DNSPolicy:          kapi.DNSClusterFirst,
+		RestartPolicy:      kapi.RestartPolicyNever,
+		ServiceAccountName: openshiftAnsibleServiceAccount,
+		//HostNetwork:        true,
+
+		Containers: []kapi.Container{
+			{
+				Name:  jobName,
+				Image: r.Image,
+				Env:   env,
+				SecurityContext: &kapi.SecurityContext{
+					RunAsUser: &runAsUser,
+				},
+				VolumeMounts: []kapi.VolumeMount{
+					{
+						Name:      "inventory",
+						MountPath: "/ansible/inventory/",
+					},
+					{
+						Name:      "sshkey",
+						MountPath: "/ansible/ssh/",
+					},
+				},
+				//Command: []string{"sleep", "1000000"},
+
+				// TODO: drop this once https://github.com/openshift/openshift-ansible/pull/6320 merges, the default run script should then work, minus the vars file :(
+				Command: []string{"ansible-playbook", "-i", "/ansible/inventory/hosts", "--private-key", "/ansible/ssh/privatekey.pem", playbookPath, "-e", "@/ansible/inventory/vars"},
```

Comment:
Does ansible allow for multiple `-e` arguments? Would it be possible to pass in the unchangeable inventory in one file and the changeable in another? I don't know yet if that is any merit to that: I was just wondering if that is even a possibility.

Reply:
I would definitely expect that it would. I'm not sure which would be easier but it wouldn't be hard to do if we want to go that route.